### ISSUE TITLE
Dependencies can be defined in `-pkg` file instead of header

### ISSUE THREAD

**Issue by Fuco1** [emacs-eask/eask#10]
I have a `smartparens-pkg.el` file which defines dependencies for a package instead of declaring it in the main file header.

Right now eask reports

```
Unmatch dependency ’dash’; add (dash "VERSION") to package-file or consider removing it
```

We should make sure to also check if a `-pkg.el` file exists and read the dependencies from there.

**Comment by jcs090218** [comment]
Hmm... Should `-pkg.el` generated by Emacs itself?

Sorry I don't know much about `-pkg.el` file. What I have in mind is the package owner should add `Package-Requires:` field in their package main file (`smartparens.el` in your case), and then hit `eask pkg-file`.

If the above statement is **true**, then I don't see why we should check `-pkg.el` file. 😅 Let me know what you think? I am not very familiar with the `pkg.el` file.

**Comment by jcs090218** [comment]
Thanks for the explanation!

My question was, "should people create/change the `-pkg.el` file manually?"

While I'm implementing command `eask pkg-file`, I feel like package author/maintainer should **NOT** manually tweak `-pkg.el` file since it's easier to change it inside the comment section, and let the program does the work (generate `-pkg.el`) for you. 😕 

Unless they recommend (official) this kind of workflow? 😅

**Comment by Fuco1** [comment]
Okay, I see what you mean now.

I'm not sure, but I suppose if Emacs sees the pkg file it will ignore the headers.  So it shouldn't be a problem to have both from `package.el`'s point of view. 

But it might be confusing for people who are used to writing the pkg files that eask is giving this warning.  Maybe we should add more explanation, including the fact that eask can manage the pkg file for you.

Because right now, I think the warning is confusing, especially if someone is just switching over and they already have the pkg file but get a warning that a dependency is not declared, while it in fact is.

I personally used `cask pkg-file` to generate it, so yea... :D but cask takes the dependencies from the cask file, not the package header.

**Comment by jcs090218** [comment]
> I'm not sure, but I suppose if Emacs sees the pkg file it will ignore the headers. So it shouldn't be a problem to have both from package.el's point of view.

I reckon supporting both methods wouldn't conflict the workflow I preferred. I was worried `-pkg.el` file will confuse new Emacsers, but I may be overthinking it.

> I personally used cask pkg-file to generate it, so yea... :D but cask takes the dependencies from the cask file, not the package header.

Does Cask have a checker? Yeah, I noticed people may not be willing to update these metadata in multiple places, (Eask, `-pkg.el`, or package headers) so I think it's good to have such a feature but not too aggressive, just a warning/tip to acknowledge the user about it. It doesn't really affect anything TBH.

> Because right now, I think the warning is confusing, especially if someone is just switching over and they already have the pkg file but get a warning that a dependency is not declared, while it in fact is.

I agreed. I have implemented this feature in #21. Would like to test it? :)

**Comment by Fuco1** [comment]
Yea, I think it works.
